tools/ratbagctl.in.in

```diff
@@ -325,6 +327,17 @@ def func_button_action_set_special(r, args):
     commit(d, args)
 
 
+def func_button_action_set_key(r, args):
+    b, p, d = find_button(r, args)
+    try:
+        key = args.target_key
+    except AttributeError:
+        pass
+    else:
+        b.key = key
+    commit(d, args)
```

Comment:
Does this work? If I understand correctly, `args.target_key` is always a string, while whatever you supply to `b.key` is supposed to be an int as I get this error:
```
> python ./build/ratbagctl Glorious button 5 action set key KEY_A
Traceback (most recent call last):
  File "/usr/lib/python3.10/site-packages/gi/_propertyhelper.py", line 401, in obj_set_property
    prop.fset(self, value)
  File "/path/to/sources/libratbag/./build/ratbagctl", line 775, in key
    key = GLib.Variant("u", key)
  File "/usr/lib/python3.10/site-packages/gi/overrides/GLib.py", line 189, in __new__
    v = creator._create(format_string, value)
  File "/usr/lib/python3.10/site-packages/gi/overrides/GLib.py", line 118, in _create
    return self._LEAF_CONSTRUCTORS[format](value)
TypeError: Must be number, not str
```

I think it would work like this, but I can't try it as it seems like the `key` button type is broken in Sinowealth driver.
```diff
diff --git a/tools/ratbagctl.in.in b/tools/ratbagctl.in.in
index ac66f8e7..e5734572 100755
--- a/tools/ratbagctl.in.in
+++ b/tools/ratbagctl.in.in
@@ -1,5 +1,3 @@
-#!/usr/bin/env python3
-#
 # Copyright 2016 Red Hat, Inc.
 #
 # Permission is hereby granted, free of charge, to any person obtaining a
@@ -41,6 +39,14 @@ button_specials_strmap = {
     **{e.name.lower().replace("_", "-"): e for e in RatbagdButton.ActionSpecial}
 }
 
+# XXX: not sure what type does it return, though we don't use type hints so it may make sense to remove `: str` too.
+def convert_str_to_evcode(s: str):
+    if not s.startswith("KEY_") and not s.startswith("BTN_"):
+        msg = "Don't know how to convert {}".format(s)
+        raise argparse.ArgumentTypeError(msg)
+
+    return evdev.ecodes.ecodes[s]
+
 
 def list_devices(r, args):
     if not r.devices:
@@ -329,12 +335,13 @@ def func_button_action_set_special(r, args):
 
 def func_button_action_set_key(r, args):
     b, p, d = find_button(r, args)
+    print(1)
     try:
-        key = args.target_key
+        code = convert_str_to_evcode(args.target_key)
     except AttributeError:
         pass
     else:
-        b.key = key
+        b.key = code
     commit(d, args)
 
 
@@ -366,11 +373,7 @@ def func_button_action_set_macro(r, args):
             t = int(s[1:])
             macro.append(RatbagdButton.Macro.WAIT, t)
         else:
-            if not s.startswith("KEY_") and not s.startswith("BTN_"):
-                msg = "Don't know how to convert {}".format(s)
-                raise argparse.ArgumentTypeError(msg)
-
-            code = evdev.ecodes.ecodes[s]
+            code = convert_str_to_evcode(s)
             if is_press:
                 macro.append(RatbagdButton.Macro.KEY_PRESS, code)
             if is_release:
```

Reply:
Not sure if this is because of my patch, but it shows key code instead of the key itself in `ratbagctl <mouse_name> info`:
```
warbling-mara - Glorious Model O
             Model: usb:258a:0036:0
 Number of Buttons: 6
    Number of Leds: 0
Number of Profiles: 1
Profile 0: (active)
  Name: n/a
  Report Rate: 1000Hz
  Resolutions:
    0: 900x900dpi (active) (default)
    1: <disabled>
    2: <disabled>
    3: <disabled>
    4: <disabled>
    5: <disabled>
    6: <disabled>
    7: 4600x4600dpi
  Button: 0 is mapped to 'button 1'
  Button: 1 is mapped to 'button 2'
  Button: 2 is mapped to 'button 3'
  Button: 3 is mapped to 'button 4'
  Button: 4 is mapped to 'button 5'
  Button: 5 is mapped to key '30'
```